bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2369,20 +2369,26 @@ private static boolean checkIsLikeArrayType(Object sourceValue, BArrayType targe
                         return true;
                     }
 
-                    if (targetTypeElementType.getTag() != TypeTags.UNION_TAG) {
+                    if (targetTypeElementType.getTag() != TypeTags.UNION_TAG &&
```

Comment:
Shall we extract `targetTypeElementType.getTag()` out to a variable?

Reply:
This piece of code has been removed with new commits